src/app/pages/home/home.jsx

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * Dependencies.
+ */
+
+var React = require('react');
+var StoreMixin = require('fluxible').StoreMixin;
+var GoogleMapsStore = require('../../maps/google-maps-store');
+var NavLink = require('../../nav/nav-link');
+var loadGoogleMaps = require('../../maps/load-google-maps');
+
+/**
+ * Home page.
+ */
+
+module.exports = React.createClass({
+  mixins: [StoreMixin],
+  statics: {
+    storeListeners: [GoogleMapsStore]
+  },
+  getInitialState: function () {
+    var venues = this.props.context.getStore('venue').getVenues();
+
+    return {
+      venues: venues
+    }
+  },
+  componentDidMount: function () {
+    var mapsIsReady = this.props.context.getStore('google-maps').isReady();
+    if (mapsIsReady) {
+      this.renderMap();
+    } else {
+      loadGoogleMaps(this.props.context);
```

Comment:
loadGoogleMaps will dispatch an action that informs the googleMapsStore than the api is ready. That will, in turn, fire an event which will tell the component to re-render itself. Both could be encountered because of in-app navigation. It's not certain whether the api is loaded or not when this component is being initialized.